cpp/libs/gst_nmos_sender_plugin/gst_nmos_sender_plugin.cpp

```diff
@@ -0,0 +1,442 @@
+                                    g_param_spec_string(name, nick, blurb, default_value,
+                                                        (GParamFlags)(G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS)));
+}
+
+/* Class initialization */
+static void gst_nmossender_class_init(GstNmossenderClass* klass)
+{
+    GObjectClass* object_class     = G_OBJECT_CLASS(klass);
+    GstElementClass* element_class = GST_ELEMENT_CLASS(klass);
+
+    gst_element_class_add_pad_template(element_class, gst_static_pad_template_get(&sink_template));
+
+    object_class->set_property = gst_nmossender_set_property;
+    object_class->get_property = gst_nmossender_get_property;
+
+    // Add properties using helper function
+    add_property(object_class, 1, "node-id", "Node ID", "The ID of the node", "0aad3458-1081-4fba-af02-a8ebd9feeae3");
+    add_property(
+        object_class, 2, "node-config-file-location", "Node Config File Location",
+        "The location of the configuration file for the node",
+        "/home/nmos/repos/nmos-sender-receiver-framework/cpp/demos/ossrf-nmos-api/config/nmos_plugin_config.json");
+    add_property(object_class, 3, "device-id", "Device ID", "The ID of the device",
+                 "b9b85f97-58db-41fe-934f-c2afbf7bd46f");
+    add_property(object_class, 4, "device-label", "Device Label", "The label to identify the device", "OSSRF Device");
+    add_property(object_class, 5, "device-description", "Device Description", "Description of the device",
+                 "OSSRF Device Description");
+    add_property(object_class, 6, "sender-id", "Sender ID", "The ID of the sender",
+                 "e543a2c1-d6a2-47f5-8d14-296bb6714ef2");
+    add_property(object_class, 7, "sender-label", "Sender Label", "Label to identify the sender",
+                 "BISECT sender video");
+    add_property(object_class, 8, "sender-description", "Sender Description",
+                 "Description string to better describe the sender", "BISECT sender video");
+    add_property(object_class, 9, "source-address", "Source Address", "The address of the source", "127.0.0.1");
+    add_property(object_class, 10, "interface-name", "Interface Name", "Name of the interface", "wlp1s0");
+    add_property(object_class, 11, "destination-address", "Destination Address", "The address of the destination",
+                 "127.0.0.1");
+    add_property(object_class, 12, "destination-port", "Destination Port", "Port of the destination", "9999");
+
+    gst_element_class_set_static_metadata(element_class, "NMOS Sender", "Sink/Network",
+                                          "Processes raw video and sends it over RTP and UDP to NMOS client",
+                                          "Luis Ferreira <luis.ferreira@bisect.pt>");
+
+    element_class->change_state = gst_nmossender_change_state;
+}
+
+/* Object initialization */
+static void gst_nmossender_init(GstNmossender* self)
+{
+    self->queue           = gst_element_factory_make("queue", "queue");
```

Comment:
The second argument of `gst_element_factory_make` specifies the name of the element within the pipeline. If this argument is set to `NULL`, GStreamer automatically assigns a guaranteed unique name, which consists of the element factory name followed by a number. If the argument is given, it will be given the name supplied.

So please pass the second argument as NULL to avoid potential issues, such as accidentally creating two elements with the same name, leading to pipeline crashes.

Change this everywhere in the code.